### PR TITLE
Make dynamic conv operand require static dim sizes, fix padding name

### PR DESCRIPTION
**Pull request by GleasonK**
Changes:

- `d_padding --> padding`
- `HLO_2DIntTensor:$d_padding --> HLO_Static2DIntTensor:$padding`

And cleanup for the two changes.

### CODE REVIEW

File: stablehlo/dialect/Base.td [hunk 1]
```diff
@@ -143,7 +143,7 @@ def HLO_Token : Type<CPred<"isa<TokenType>($_self)">, "token">;
 def HLO_IntTensor : RankedTensorOf<[HLO_Int]>;
 
 // Any integer tensor type with rank 2.
-def HLO_2DIntTensor : TensorRankOf<[HLO_Int], [2]>;
+def HLO_Static2DIntTensor : RankedTensorOf<[HLO_Int], [HasStaticShapePred, HasAnyRankOfPred<[2]>], "statically shaped 2-dimensional integer tensor">;
```
Comment (ghpvnist) [inline stablehlo/dialect/Base.td:146]: Let's move this to around L260. There's a place for static types there.

File: stablehlo/dialect/VhloOps.td [hunk 2]
```diff
@@ -453,13 +453,13 @@ def VHLO_DynamicConvOpV1 : VHLO_Op<"dynamic_conv_v1", "0.9.0", "0.19.0"> {
   let results = (outs VHLO_AnyType:$result);
 }
 
-// `d_padding` should be used instead of `padding` for dynamic convolution, so
-// `padding` is removed for clarity.
+// Padding should be specified as an operand only, not an attribute.
+// Remove `d_padding` and convert `padding` to an operand.
```
Comment (ghpvnist) [inline stablehlo/dialect/VhloOps.td:457]: Not sure what's more accurate; your call:
"Remove `padding` and rename `d_padding` to `padding`?"

**Review by ghpvnist — APPROVED** [review]
LGTM with comments.

**Review by sdasgup3 — APPROVED** [review]

File: stablehlo/dialect/VhloOps.td (hunk 2 above)
Comment (GleasonK) [reply stablehlo/dialect/VhloOps.td:457]: That's what I had written first, then I changed it.. they're equally clunky IMO 😓 . Going to leave as-is

**Pull request merged**